### PR TITLE
[WIP][docs] Document VPN Automation

### LINKED ISSUE #44: Step by step tutorial to explain how to use the VPN-client automation feature
First of all, you should be able to replicate this feature in your own development environment.

How to do that is roughly explained in this mailing list thread: https://groups.google.com/d/msg/openwisp/xVZR7vy1kck/OQzzD8OfAgAJ

Take notes while you try to do it.

Once you are able to have a device automatically connect to the VPN after registration, use your notes to create a new page in the user documentation. 

Document the following concepts

- VPN automation what's for
- VPN templates
- explain what happens when a VPN template is flagged as default
- explain the "auto client certificates" feature
- explain how to install a VPN server with ansible and how to import the VPN server configuration into OpenWISP (explain also that this is not automated yet)

Once ready, send a pull request and follow up on our reviews.

### PR DESCRIPTION
**Pull request by wizanyx**
Made a step by step tutorial to explain how to use the VPN-client automation feature.
Closes #44

### CODE REVIEW

File: user/vpn.rst [hunk 1]
```diff
@@ -0,0 +1,143 @@
+VPN-Client Automation
+=====================
+
+
+Installing VPN Server and importing the VPN configuration
+---------------------------------------------------------
+We will be installing VPN Server using `this ansible role <https://github.com/Stouts/Stouts.openvpn>`_
```
Comment (nemesisdesign) [inline user/vpn.rst]: leave a blank line after the title, applies for other titles as well

File: user/vpn.rst [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+and  then import the VPN configuration into OpenWISP.
```
Comment (nemesisdesign) [inline user/vpn.rst]: I would specify that one can also install the VPN server in another way if they feel more comfortable, the important thing is that they get the VPN server running and then they import it into OpenWISP

File: user/vpn.rst [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+**Importing the CA**:
+
+On your openwisp dashboard go to ``/admin/pki/ca/add/``
+
+In **Operation Type** choose :guilabel:`Import Existing`
+
+Get your CA Ceritficate from ``ca.crt`` file and Private Key from
+``private/ca.key`` and then enter them in the respective fields.
+
```
Comment (nemesisdesign) [inline user/vpn.rst]: remove blank line

File: user/vpn.rst [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+On your openwisp dashboard go to ``/admin/config/vpn/add/``
+
+In **Host** enter you Server IP, in **Certification Authority** select
+the CA you created and in **X509 Certificate** select the certificate you
+created.
+
+Now under **Configuration**, open **Configuration Menu** and deselect
+Property :guilabel:`Files`. For **VPN1** Change
+:guilabel:`Server (Bridged)` to :guilabel:`Server (Routed)`.
```
Comment (nemesisdesign) [inline user/vpn.rst]: OpenWISP can work with both modes, so I would make explicit that in this guide we're showing how to do the Routed server (Layer3 VPN) but also mention that the process to setup a Bridged server (Layer2 VPN) is identical.

File: user/vpn.rst [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+Change the rest of the Configuration of the VPN according to the
+configuration in ``/etc/openvpn/server.conf``
+
+.. Tip::
+    You can check if your VPN Configuration is similar to the
+    ``server.conf`` file using the **Preview Configuration** option
+    at the Top.
```
Comment (nemesisdesign) [inline user/vpn.rst]: Ok now we need to explain how to create the VPN client template.

I would add the suggestion to set the VPN client as default template and then show how a newly registered device gets the template installed automatically and the VPN client is brought up automatically (showing screenshos)

**Comment by Weirdo914** [comment]
I was going to add the images at last after I had done all the writing stuff.

File: user/vpn.rst [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+
+VPN Templates
+-------------
+
+Creating a VPN Template
+~~~~~~~~~~~~~~~~~~~~~~~
+
+On your openwisp dashboard go to ``/admin/config/template/add/``
```
Comment (nemesisdesign) [inline user/vpn.rst]: missing dot

File: user/vpn.rst [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+
+You can further toggle `Enabled by default <#default-templates>`_
+and `Auto certificate <#auto-client-certificates>`_
+options according to your needs.
+
+Save the template. You can now tweak the Client VPN configuration.
+Now can add the template to your devices.
+
+Auto Client Certificates
+~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (nemesisdesign) [inline user/vpn.rst]: move this section after default templates

File: user/vpn.rst [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+
+**Option**: ``Auto certificate``
+
+**Default**: ``True``
+
+Auto Client Certificates feature allows you to automatically generate
+generate client certificates for your Device.
```
Comment (nemesisdesign) [inline user/vpn.rst]: Replace `Auto Client Certificates feature allows you` with `This feature allows`.

`for your Device` > `for devices which have this template assigned.`

File: user/vpn.rst [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+Default Templates
+~~~~~~~~~~~~~~~~~
+
+**Option**: ``Enabled by default``
+
+**Default**: ``False``
+
+Default templates are automatically added to newly created devices of
+the organisation of the template. If no organisation is specified, the
+template is added to all the organisations.
```
Comment (nemesisdesign) [inline user/vpn.rst]: `to all devices of all the organisations`

File: user/vpn.rst [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+
+Change the rest of the Configuration of the VPN according to the
+configuration in ``/etc/openvpn/server.conf``
+
+.. Tip::
+    You can check if your VPN Configuration is similar to the
+    ``server.conf`` file using the **Preview Configuration** option
+    at the Top.
+
+VPN Templates
```
Comment (nemesisdesign) [inline user/vpn.rst]: Configuring VPN Clients

File: user/vpn.rst [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+-------------
+
+Creating a VPN Template
```
Comment (nemesisdesign) [inline user/vpn.rst]: `Creating a VPN client Template`

File: user/vpn.rst [hunk 12]
```diff
@@ -0,0 +1,190 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~
+
+On your openwisp dashboard go to ``/admin/config/template/add/``
+
+Change **Type** to :guilabel:`VPN-client`.
+For **VPN** select the VPN you created.
+
+You can further toggle `Enabled by default <#default-templates>`_
+and `Auto certificate <#auto-client-certificates>`_
+options according to your needs.
```
Comment (nemesisdesign) [inline user/vpn.rst]: for management VPNs, we recommend enabling both.

**Pull request merged**